api/src/pcapi/core/educational/api/stock.py

```diff
@@ -82,7 +83,21 @@ def edit_collective_stock(
     if booking_limit is None:
         check_booking_limit_datetime = stock.bookingLimitDatetime
 
-    offer_validation.check_booking_limit_datetime(stock, check_beginning, check_booking_limit_datetime)
+    booking_status = [booking.status for booking in stock.collectiveBookings]
```

Comment:
```suggestion
    current_booking = [booking for booking in stock.collectiveBookings if booking.status != CollectiveBookingStatus.CANCELLED] or None
```
ça simplifie la suite

Reply:
pas sur de comprendre cette logique en tout cas ca fait tout peter

Reply:
`[booking.status for booking in stock.collectiveBookings if booking.status != CollectiveBookingStatus.CANCELLED]`

Reply:
```current_bookings = [booking for booking in stock.collectiveBookings if booking.status != CollectiveBookingStatus.CANCELLED]

current_booking = current_bookings[0] if current_bookings or None
```